src/main/java/io/horrorshow/codey/time/DiscordTimezone.java

```diff
@@ -0,0 +1,135 @@
+package io.horrorshow.codey.time;
+
+import io.horrorshow.codey.discordutil.DiscordUtils;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.Arrays;
+
+@Service
+public class DiscordTimezone extends ListenerAdapter {
+    private final DiscordUtils utils;
+
+    public DiscordTimezone(@Autowired JDA jda, @Autowired DiscordUtils utils) {
+        this.utils = utils;
+
+        jda.addEventListener(this);
+    }
+
+    private final String emoji = "⏰";
+    private final String deleteEmoji = "\uD83D\uDDD1";
+
+    @Override
+    public void onGuildMessageReceived(@NotNull GuildMessageReceivedEvent event) {
+        if (!event.getAuthor().isBot()) {
+            onMessage(event.getMessage());
+        }
+    }
+
+    private void onMessage(Message message) {
+        String rawContent = message.getContentRaw();
+        String[] rawContentParts = rawContent.split("\\s+");
+
+        if(rawContentParts.length == 0)
+            return;
+
+        Object[] result = time(rawContentParts);
+
+        if(!((boolean) result[0])) {
+            if(result[1] != null && (boolean) result[1])
+            {
+                message.addReaction(emoji).complete();
+                message.reply(new EmbedBuilder()
+                        .setColor(utils.getColor())
+                        .setTitle("Did you know that I can provide time conversions automatically?")
+                        .addField("Here is a tip:",
+                                "You can use regions from the IANA Time Zone Database (TZDB). " +
+                                "This has region IDs of the form `{area}/{city}`, such as `Europe/Paris` or `America/New_York`", false)
+                        .build()).complete().addReaction(deleteEmoji).complete();
+            }
+            return;
+        }
+
+        OffsetDateTime timeStamp = (OffsetDateTime) result[3];
+
+        message.reply(new EmbedBuilder()
+                .setFooter("Time in your timezone: ")
+                .setTimestamp(timeStamp)
+                .setColor(utils.getColor())
+                .build()).mentionRepliedUser(false).complete();
+    }
+
+    private Object[] time(String[] rawContentParts) {
+        boolean returnValue = false;
+        Object[] array = new Object[4];
+
+        int index = -1;
+
+        for(int i = 0; i < rawContentParts.length; i++)
+        {
+            String s = rawContentParts[i];
+            if(!s.contains(":"))
```

Comment:
You can use a regex to find matches that look like 10:47, I suggest creating a collection of regex (instead of making one really long one which noone can read) and trying all of them on a message, so you can match various ways to write time and it's easy to add new ones later (for example for 24h or 12pm/am variations or with seconds etc.
For example `([0-9]|0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]` for a 24h format. There are some samples here: [https://stackoverflow.com/questions/7536755/regular-expression-for-matching-hhmm-time-format/7536768](stackoverflow)

To find all occurences that match a regex in a string it'll look something like:
```java
var pattern = Pattern.compile("your regex");
var matcher = pattern.matcher("the discord message as string");
while (matcher.find()) {
  // matcher.group() returns the matched string, in your case the extracted time
  // matcher.start() returns the start index of the match
  // matcher.end() returns the end index of the match
}
```

Also add unit tests to visualise the behaviour, basically showing with string A it matches B for various sample strings.